validator/client/validator_attest.go

```diff
@@ -87,17 +87,9 @@ func (v *validator) SubmitAttestation(ctx context.Context, slot uint64, pubKey [
 		return
 	}
 
-	var history *slashpb.AttestationHistory
+	attesterHistory := v.attesterHistoryByPubKey[pubKey]
```

Comment:
This also needs a lock if called concurrently. Consider using sync.RWMutex and making this an RLock() and RUnlock() call